Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabUndoHelpers.cpp

```diff
@@ -52,25 +52,38 @@ namespace AzToolsFramework
             }
 
             void AddEntity(
-                const PrefabDomValue& parentEntityDomBeforeAddingEntity,
-                const PrefabDomValue& parentEntityDomAfterAddingEntity,
                 const PrefabDomValue& newEntityDom,
                 AZ::EntityId entityId,
-                AZ::EntityId parentEntityId,
                 TemplateId templateId,
                 UndoSystem::URSequencePoint* undoBatch)
             {
-                // Create undo node for adding entity to the appropriate prefab template.
                 PrefabUndoAddEntity* addEntityUndoState = aznew PrefabUndoAddEntity("Undo Adding Entity");
                 addEntityUndoState->SetParent(undoBatch);
                 addEntityUndoState->Capture(newEntityDom, entityId, templateId);
                 addEntityUndoState->Redo();
+            }
 
-                // Create undo node to account for changes to parent entity due to adding a new entity under it. Currently only the
-                // EditorEntitySortComponent get modified on the parent but more things can change in the future too.
-                PrefabUndoEntityUpdate* state = aznew PrefabUndoEntityUpdate("Undo parent entity update");
+            void RemoveEntities(
+                const AZStd::vector<PrefabDom>& removedEntityDoms,
```

Comment:
This should be const PrefabDomValue& so that you don't have to create a copy of the dom.

Reply:
Yes. I think `AZStd::vector<const PrefabDomValue*>` should work here. It is currently using the DOM value (to remove entity DOM) before going into the parent update step. So, it makes sense to avoid the unnecessary copying.